plugins/quota/validate.go

```diff
@@ -3,44 +3,66 @@ package quota
 import (
 	"fmt"
 
+	daemon "github.com/gluster/glusterd2/glusterd2/daemon"
 	"github.com/gluster/glusterd2/glusterd2/volume"
 	"github.com/gluster/glusterd2/glusterd2/xlator"
+	"github.com/gluster/glusterd2/pkg/errors"
 )
 
 const (
-	quotaEnabledKey = "features.quota"
+	quotaEnabledKey = "quota.enable"
 )
 
 func validateOptions(v *volume.Volinfo, key string, value string) error {
 
-	if isQuotaEnabled(v) == false {
-		err := fmt.Errorf("Quota not enabled to set this value: '%s'", key)
-		return err
+	// Check if volume is started
+	if v.State != volume.VolStarted {
+		return errors.ErrVolNotStarted
 	}
 
 	switch key {
-	case "deem-statfs":
+	case "enable":
+		switch value {
+		case "on":
+			if isQuotaEnabled(v) {
+				return errors.ErrProcessAlreadyRunning
+			}
+		case "off":
+			if !isQuotaEnabled(v) {
+				return errors.ErrProcessNotRunning
+			}
+		}
 		return nil
+	case "deem-statfs":
+		fallthrough
 	case "hard-timeout":
-		return nil
+		fallthrough
 	case "soft-timeout":
-		return nil
+		fallthrough
 	case "alert-time":
-		return nil
+		fallthrough
 	case "default-soft-limit":
-		return nil
+		if isQuotaEnabled(v) == false {
+			return fmt.Errorf("Quota not enabled to set this value: '%s'", key)
+		}
 	default:
-		err := fmt.Errorf("'%s' is not a valid quota option", key)
-		return err
+		return fmt.Errorf("'%s' is not a valid quota option", key)
 	}
+
+	return nil
 }
 
 func isQuotaEnabled(v *volume.Volinfo) bool {
-	val, exists := v.Options[quotaEnabledKey]
-	if exists && val == "on" {
-		return true
+	pid, err := daemon.ReadPidFromFile("/usr/local/var/run/glusterd2/quotad.pid")
```

Comment:
I'm sorry, I should've been clearer - I expect two separate functions to exist:

* `isQuotaEnabled` checks if quota is enabled
* `isQuotadRunning` checks if the daemon is actually running

You may have to use both of these functions.

In `isQuotadRunning()`, do not read the hard-coded pidfile directly. Instead, use the `PidFile()` method defined on the quota type that implements the daemon interface,

Reply:
done